src/room/CallEventAudioRenderer.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2024 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only
+Please see LICENSE in the repository root for full details.
+*/
+
+import { ReactNode, useDeferredValue, useEffect, useRef } from "react";
+import { useObservableEagerState } from "observable-hooks";
+
+import {
+  playReactionsSound,
+  soundEffectVolumeSetting as effectSoundVolumeSetting,
+  useSetting,
+} from "../settings/settings";
+import { CallViewModel } from "../state/CallViewModel";
+import enterCallSoundMp3 from "../sound/join_call.mp3";
+import enterCallSoundOgg from "../sound/join_call.ogg";
+import leftCallSoundMp3 from "../sound/left_call.mp3";
+import leftCallSoundOgg from "../sound/left_call.ogg";
+
+// Do not play any sounds if the participant count has exceeded this
+// number.
+export const MAX_PARTICIPANT_COUNT_FOR_SOUND = 8;
+
+export function CallEventAudioRenderer({
+  vm,
+}: {
+  vm: CallViewModel;
+}): ReactNode {
+  const [shouldPlay] = useSetting(playReactionsSound);
+  const [effectSoundVolume] = useSetting(effectSoundVolumeSetting);
+  const memberIds = useObservableEagerState(vm.userMediaIds);
+  const previousMembers = useDeferredValue(memberIds);
+  const callEntered = useRef<HTMLAudioElement>(null);
+  const callLeft = useRef<HTMLAudioElement>(null);
```

Comment:
As discussed already i think it would be cool to make the join/leave an observable as well.
This could be built with a `scan`.
sth like
```
let memberChanges = memberIds.pipe(scan((prev, ids)=>{
    const left = prev.ids.filter((id)=>ids.includes(id));
    const joined = ids.filter((id)=>prev.ids.includes(id));
    return {ids, joined, left}
}));
```
than this could be used like an event emitter that emits on join and leave:
```
memberChanges.pipe(
  map((m)=>m.left),
  filter((left)=>left.length === 0),
).subscribe((left)=>"play sond on leave")
```